src/tringa/fetch.py

```diff
@@ -141,41 +143,49 @@ def _parse_xml_file(
+                for result in test_case.result or [empty_result]:
+                    text = result.text
+                    if text and len(text) > MAX_TEST_OUTPUT_LENGTH:
+                        debug(
+                            f"Truncating {file} output from {len(text)} to {MAX_TEST_OUTPUT_LENGTH}"
+                        )
+                        text = text[:MAX_TEST_OUTPUT_LENGTH] + "...<truncated by tringa>"
+                    yield TestResult(
+                        repo=run.repo,
+                        artifact=artifact_name,
+                        run_id=run.id,
+                        branch=run.branch,
+                        sha=run.sha,
+                        pr=pr.number if pr else None,
+                        pr_title=pr.title if pr else None,
+                        file=file.name,
+                        suite=test_suite.name,
+                        suite_time=(
+                            datetime.fromisoformat(test_suite.timestamp)
+                            if test_suite.timestamp
+                            else None
+                        ),
+                        suite_duration=test_suite.time,
+                        name=test_case.name,
+                        classname=test_case.classname or "",
+                        flaky=False,
+                        duration=test_case.time,
+                        passed=test_case.is_passed,
+                        skipped=test_case.is_skipped,
+                        message=result.message,
+                        text=text,
                     )
-                    text = text[:MAX_TEST_OUTPUT_LENGTH] + "...<truncated by tringa>"
-                yield TestResult(
-                    repo=run.repo,
-                    artifact=artifact_name,
-                    run_id=run.id,
-                    branch=run.branch,
-                    sha=run.sha,
-                    pr=pr.number if pr else None,
-                    pr_title=pr.title if pr else None,
-                    file=file.name,
-                    suite=test_suite.name,
-                    suite_time=(
-                        datetime.fromisoformat(test_suite.timestamp)
-                        if test_suite.timestamp
-                        else None
-                    ),
-                    suite_duration=test_suite.time,
-                    name=test_case.name,
-                    classname=test_case.classname or "",
-                    flaky=False,
-                    duration=test_case.time,
-                    passed=test_case.is_passed,
-                    skipped=test_case.is_skipped,
-                    message=result.message,
-                    text=text,
-                )
+    except (xml.etree.ElementTree.ParseError, xml.sax.SAXParseException) as e:
+        warn(f"Skipping malformed XML file {file}: {e}")
+        return
+    except Exception as e:
+        warn(f"Skipping XML file {file} due to parsing error: {e}")
```

Comment:
You're sure that converting all exceptions to warnings here is the right thing to do, right? I think I'd prefer not to say "parsing error" since it could literally be anything, e.g. a typo'd variable name.

Reply:
I made this change because I introduced a bug into the temporal testrunner causing malformed XML in our junit reports. Currently `tringa` will fail without any processing if even one of the junit reports is improper XML. _I think_ this is a better experience because `tringa` will try to parse the rest if one fails with an appropriate warning log of what caused the skip of a junit report.

Reply:
OK, sounds good! The error message isn't likely to cause any confusion.